devito/function.py

```diff
@@ -273,15 +278,22 @@ def _initialize_derivatives(self):
                                  'the first cross derivative wrt. the '
                                  '%s and %s dimensions' %
                                  (dim.name, dim2.name)))
-
+                derivatives += ('d%s%s' % (dim.name, dim2.name), )
                 # Second cross derivative
-                dx2y2 = partial(second_cross_derivative, dims=(dim, dim2),
-                                order=self.space_order)
-                setattr(self.__class__, 'd%s2%s2' % (dim.name, dim2.name),
-                        property(dx2y2, 'Return the symbolic expression for '
-                                 'the second cross derivative wrt. the '
-                                 '%s and %s dimensions' %
-                                 (dim.name, dim2.name)))
+                if self.space_order >= 2:
+                    dx2y2 = partial(second_cross_derivative, dims=(dim, dim2),
+                                    order=self.space_order)
+                    setattr(self.__class__, 'd%s2%s2' % (dim.name, dim2.name),
+                            property(dx2y2, 'Return the symbolic expression for '
+                                     'the second cross derivative wrt. the '
+                                     '%s and %s dimensions' %
+                                     (dim.name, dim2.name)))
+                    derivatives += ('d%s2%s2' % (dim.name, dim2.name), )
+        if len(derivatives) > 0:
+            debug("Spatial derivatives shortcuts generated for " + str(self) +
+                  " are: \n %s" % list(derivatives))
+        else:
+            debug("No spatial derivatives shortcuts generated for " + str(self))
```

Comment:
Perhaps these 4 lines can be turned into:
``debug("Spatial derivatives supported by %s are: %s" % (self, list(derivatives)))``
the empty list simply means no spatial derivatives were generated

Reply:
What I had originaly but looked weird, can put it back.